src/pipeline/rackhd/ova/deploy.sh

```diff
@@ -0,0 +1,470 @@
+    # OVA_INTERNAL_IP, eth1 IP of ova
+    ssh-keygen -f "$HOME/.ssh/known_hosts" -R $OVA_INTERNAL_IP
+
+    echo "*****************************************************************************************************"
+    echo "End to deploy rackhd ova."
+    echo "*****************************************************************************************************"
+}
+
+###################################
+#
+# Wait for rackhd services
+#
+#################################
+waitForAPI() {
+  service_normal_sentence="No auth token"
+  timeout=0
+  maxto=60
+  set +e
+  url=http://$OVA_INTERNAL_IP:8080/api/2.0/nodes
+  while [ ${timeout} != ${maxto} ]; do
+    api_test_result=`curl ${url}`
+    echo $api_test_result | grep "$service_normal_sentence" > /dev/null  2>&1
+    if [ $? = 0 ]; then
+      echo "[Debug] successful.        in this retry time: OVA ansible returns: $api_test_result"
+      break
+    fi
+    sleep 10
+    timeout=`expr ${timeout} + 1`
+  done
+  set -e
+  if [ ${timeout} == ${maxto} ]; then
+    echo "Timed out waiting for RackHD API service (duration=`expr $maxto \* 10`s)."
+    exit 1
+  fi
+}
+
+###################################
+#
+# Modify the RackHD ova vm
+#
+#################################
+configRackHDOvaVm(){
+    echo "*****************************************************************************************************"
+    echo "Config RackHD ova vm for function test"
+    echo "*****************************************************************************************************"
+
+    # config the OVA for post test
+    pushd $ON_BUILD_CONFIG_DIR/src/pipeline/rackhd/ova/ansible
+      echo "ova-post-test ansible_host=$OVA_INTERNAL_IP ansible_user=$OVA_USER ansible_ssh_pass=$OVA_PASSWORD ansible_become_pass=$OVA_PASSWORD" > hosts
+      cp -f ${ON_BUILD_CONFIG_DIR}/resources/pipeline/rackhd/ova/config.json .
+
+      if [ -z "${EXTERNAL_VSWITCH}" ]; then
+        ansible-playbook -i hosts main.yml --extra-vars "ova_gateway=$OVA_GATEWAY ova_net_interface=$OVA_NET_INTERFACE dns_server_ip=$DNS_SERVER_IP"  --tags "config-gateway"
+      fi
+      ansible-playbook -i hosts main.yml --tags "before-test" --extra-vars "ova_gateway=$OVA_GATEWAY"
+    popd
+}
+
+portForwarding(){
+    # forward ova to localhost
+    # according to vagrant/mongo/config.json and cit/fit config
+    socat TCP4-LISTEN:9091,forever,reuseaddr,fork TCP4:$1:5672 &
+    socat TCP4-LISTEN:9090,forever,reuseaddr,fork TCP4:$1:8080 &
+    socat TCP4-LISTEN:9092,forever,reuseaddr,fork TCP4:$1:9080 &
+    socat TCP4-LISTEN:9093,forever,reuseaddr,fork TCP4:$1:8443 &
+    ssh-keygen -f "$HOME/.ssh/known_hosts" -R [localhost]:2222
+    socat TCP4-LISTEN:2222,forever,reuseaddr,fork TCP4:$1:22 &
+    socat TCP4-LISTEN:37017,forever,reuseaddr,fork TCP4:$1:27017 &
+    echo "Finished ova -> localhost port forwarding"
+    echo "5672->9091"
+    echo "8080->9090"
+    echo "9080->9092"
+    echo "8443->9093"
+    echo "22->2222"
+    echo "27017->37017"
```

Comment:
I remember we discussed about the "port forwarding"
And there is a proposal to change "RackHD/test/config/stack_config.json"
From:
```
"ova": {
    "rackhd_host": "localhost",
    "install-config": {
      "ports": {
        "mongo_port": 37017,
        "http": 9090,
        "https": 9093,
        "amqp_ssl": 9091,
        "amqp_nossl": 5671,
        "ssh": 2222
      }
    }
``` 
to
```
"ova": {
    "rackhd_host": "ova_ip",
    "install-config": {
      "ports": {
        "mongo_port": 27017,
        "http": 8080,
        "https": 8443,
        "amqp_ssl": 5672,
        "amqp_nossl": 5671,
        "ssh": 22
      }
    }
```
Or is it already tested and failed?

Reply:
the change will affect prod jenkins.
So I want to do this when refactor all done and goes online.
Otherwise  for apply new "RackHD/test/config/stack_config.json the scripts of prod jenkins need be changed

Reply:
How about put a specific config file under resources and cover the RackHD/test/config/stack_config.json?

Reply:
+1 nice idea